packages/arb-token-bridge-ui/src/components/TransferPanel/AdvancedSettings.tsx

```diff
@@ -100,15 +88,23 @@ export const AdvancedSettings = () => {
   const [inputLocked, setInputLocked] = useState(true)
   const [warning, setWarning] = useState<string | null>(null)
 
-  const { destinationAddress, setDestinationAddress } =
-    useDestinationAddressStore()
+  const [
+    { destinationAddress: destinationAddressFromQueryParams },
+    setQueryParams
+  ] = useArbQueryParams()
+  const [destinationAddress, setDestinationAddress] = useState<
+    string | undefined
+  >(destinationAddressFromQueryParams)
+  const debouncedDestinationAddress = useDebounce(destinationAddress, 100)
   const { destinationAddressError: error } = useDestinationAddressError()
 
   useEffect(() => {
-    // Initially hide for EOA
-    setAdvancedSettingsCollapsed(isEOA)
-    // Initially lock for EOA
-    setInputLocked(isEOA)
+    // Initially hide for EOA and if destination address query param is empty
+    setAdvancedSettingsCollapsed(
+      isEOA && typeof destinationAddress === 'undefined'
+    )
+    // Initially lock for EOA and if destination address query param is empty
+    setInputLocked(isEOA && typeof destinationAddress === 'undefined')
   }, [isEOA])
```

Comment:
missing dep

Reply:
We don't want it there, it would trigger on each destination address change

Reply:
That can cause some weird effects. Is it a big issue to call it on every keypress here, it should not be a big change in performance

Reply:
We only want to do it when page loads and when we load in the wallet type. If we do it whenever `destinationAddress` changes then advanced settings will collapse every time `destinationAddress` is set to undefined.

Reply:
https://github.com/OffchainLabs/arbitrum-token-bridge/pull/2094/commits/9b3a446f8b2414af71d3ec3f1c56ac01231a2b9e

How about this fix? It works and fixes dependencies array.

Reply:
That would work. If we only want on the initial render, then we can omit completely the dependency array. But it felt weird to only have some of the dependencies.

Reply:
Yeah completely omitting wouldn't work because we need to wait for the wallet type to load